hada-manager/src/common/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from "react";
+
+function getSavedValue(key:string,initialValue:any){
+    let storageItem = localStorage.getItem(key);
+
+    if (storageItem !== null)  JSON.parse(storageItem);
+
+    if (initialValue instanceof Function) return initialValue();
+
+    return initialValue;
+}
+
+export default function useLocalStorage(key: string, initialValue: any) {
+    const [value, setValue] = useState(() => {
+        return getSavedValue(key, initialValue);
+    });
```

Comment:
no need to pass a function inside the useState intiail value, you can just call it instead

```suggestion
    const [value, setValue] = useState(getSavedValue(key, initialValue));
```